research-prototype/ipa/ipae2e.py

```diff
@@ -0,0 +1,229 @@
+"""
+IPA End-to-end MPC in MP-SPDZ
+"""
+import operator
+from enum import IntEnum
+
+from Compiler.library import print_ln, tree_reduce
+from Compiler.types import Array, Matrix, sint, sintbit
+
+
+class Columns(IntEnum):
+    """
+    These are the Columns in the reports Matrix
+    """
+
+    MATCHKEY = 0
+    IS_TRIGGER = 1
+    VALUE = 2
+    BREAKDOWN_KEY = 3
+
+
+# BUG this shouldn't be needed
```

Comment:
this works for me without this. i'm going to delete it unless someone says we need it.